apps/swirl-docs/src/components/CommandMenu/CommandPalette.tsx

```diff
@@ -0,0 +1,109 @@
+import { ALGOLIA_INDEX } from "@swirl/lib/search";
+import classNames from "classnames";
+import { Command } from "cmdk";
+import { useEffect, useRef, useState } from "react";
+import { InstantSearch, SearchBox } from "react-instantsearch-hooks-web";
+import { searchClient } from "../Search/Algolia";
+import { useRouter } from "next/router";
+import { CustomHits } from "./CustomHits";
+import { SearchClient } from "algoliasearch/lite";
+import commandPaletteObserver, {
+  CommandPaletteObserver,
+} from "@swirl/lib/search/commandPaletteObserver";
+
+const algoliaClient: SearchClient = {
+  ...searchClient,
+  search(requests: any) {
+    if (requests.every(({ params }: any) => !params.query)) {
+      return Promise.resolve({
+        results: requests.map(() => ({
+          hits: [],
+          nbHits: 0,
+          nbPages: 0,
+          page: 0,
+          processingTimeMS: 0,
+          hitsPerPage: 0,
+          exhaustiveNbHits: false,
+          query: "",
+          params: "",
+        })),
+      });
+    }
+
+    return searchClient.search(requests);
+  },
+};
+
+export const CommandPalette = () => {
+  const [open, setOpen] = useState(false);
+  const router = useRouter();
+  const containerRef = useRef(null);
+
+  const onOpenStateUpdate: CommandPaletteObserver = (isOpen: boolean) => {
+    setOpen(isOpen);
```

Comment:
**suggestion:** Could we restore focus after closing the search? W3C suggestion: When a dialog closes, focus returns to the element that invoked the dialog.